core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestClient.scala

```diff
@@ -289,6 +337,14 @@ private[deploy] class StandaloneRestClient extends Logging {
   private def handleUnexpectedRestResponse(unexpected: SubmitRestProtocolResponse): Unit = {
     logError(s"Error: Server responded with message of unexpected type ${unexpected.messageType}.")
   }
+
+  private def handleSubmitRestConnectionException(url: String): Boolean = {
+    if (!lostMasters.contains(url)) {
+      logWarning(s"Unable to connect to server ${url}.")
+      lostMasters += url
+    }
+    lostMasters.size >= masters.size
```

Comment:
same here, I don't think this proxy makes sense since we never remove from `lostMasters`